Document ListBuckets extension (#4563)
cloudflare/cloudflare-docs: content/r2/platform/s3-compatibility/extensions.md

### MERGE metadata directive

The `x-amz-metadata-directive` allows a `MERGE` value, in addition to the standard `COPY` and `REPLACE` options. When used, `MERGE` is a combination of `COPY` and `REPLACE`, which will `COPY` any metadata keys from the source object and `REPLACE` those that are specified in the request with the new value. You cannot use `MERGE` to remove existing metadata keys from the source — use `REPLACE` instead.

## `ListBuckets`

> `ListBuckets` supports all the same search parameters as `ListObjectsV2` in R2 because some customers may have more than 1,000 buckets. Since tooling may not expose a way to set these search parameters, these values may also be sent in via headers. Values in headers take precedence over the search parameters.

Reviewer: @vlovich what 'tooling' is being described here?

Author: Existing S3 libraries/utilties etc.

> `ListBuckets` supports all the same search parameters as `ListObjectsV2` in R2 because some customers may have more than 1,000 buckets. Since tooling may not expose a way to set these search parameters, these values may also be sent in via headers. Values in headers take precedence over the search parameters.

Reviewer: ```suggestion
`ListBuckets` supports all the same search parameters as `ListObjectsV2` in R2 because some customers may have more than 1,000 buckets. Because tooling, like existing S3 libraries, may not expose a way to set these search parameters, these values may also be sent in via headers. Values in headers take precedence over the search parameters.
```

### MERGE metadata directive

The `x-amz-metadata-directive` allows a `MERGE` value, in addition to the standard `COPY` and `REPLACE` options. When used, `MERGE` is a combination of `COPY` and `REPLACE`, which will `COPY` any metadata keys from the source object and `REPLACE` those that are specified in the request with the new value. You cannot use `MERGE` to remove existing metadata keys from the source — use `REPLACE` instead.

## ListBuckets

> `ListBuckets` supports all the same search parameters as `ListObjectsV2` in R2 because some customers may have more than 1,000 buckets.
> Since tooling may not expose a way to set these search parameters, these values may also be sent in via headers.
> Values in headers take precedence over the search parameters.

Reviewer: ```suggestion
`ListBuckets` supports all the same search parameters as `ListObjectsV2` in R2 because some customers may have more than 1,000 buckets. Since tooling may not expose a way to set these search parameters, these values may also be sent in via headers. Values in headers take precedence over the search parameters.
```

## ListBuckets

In R2 ListBuckets supports all the same search parameters as `ListObjectsV2` because some customers may have more than 1000 buckets.
Since tooling may not expose a way to set these search parameters, these values may also be sent in via headers.
Values in headers take precedence over the search parameters.

{{<table-wrap>}}
| Search parameter     | HTTP Header             | Meaning                                                           |
|--------------------- | ----------------------- | ----------------------------------------------------------------- |
| `prefix`             | `cf-prefix`             | Show buckets with this prefix only.                               |
| `start-after`        | `cf-start-after`        | Show buckets whose name appears lexicographically in the account. |
| `continuation-token` | `cf-continuation-token` | Resume listing from a previously returned continuation token.     |

> | `max-keys`           | `cf-max-keys`           | Return this maximum number of buckets. Default and max is 1000.   |

Reviewer: ```suggestion
| `max-keys`           | `cf-max-keys`           | Return this maximum number of buckets. Default and max is `1000`.   |
```

{{</table-wrap>}}

The XML response contains a NextContinuationToken and IsTruncated elements as appropriate. Since these may not be accessible
from existing S3 APIs, these are also available in response headers:

{{<table-wrap>}}
| XML Response Element    | HTTP Response Header         | Meaning                                                                                      |
| ----------------------- | ---------------------------- | -------------------------------------------------------------------------------------------- |

> | `IsTruncated`           | `cf-is-truncated`            | This is set to "true" if the list of buckets returned is not all the buckets on the account. |

Reviewer: ```suggestion
| `IsTruncated`           | `cf-is-truncated`            | This is set to `true` if the list of buckets returned is not all the buckets on the account. |
```

> | `NextContinuationToken` | `cf-next-continuation-token` | This is set to continuation token to pass on a subsequent ListBuckets to resume the listing. |

Reviewer: ```suggestion
| `NextContinuationToken` | `cf-next-continuation-token` | This is set to continuation token to pass on a subsequent `ListBuckets` to resume the listing. |
```
